rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -927,4 +943,77 @@ public void removeThirdPartyCredential(String key) throws NotConnectedException,
             throw new PermissionException(e);
         }
     }
+
+    @Override
+    public Page<TaskId> getTaskIds(String taskTag, long from, long to, boolean mytasks, boolean running,
+            boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedException, PermissionException {
+        RestPage<TaskStateData> page = null;
+        try {
+            page = restApi().getTaskStates(sid, from, to, mytasks, running, pending, finished, offset, limit);
+        } catch (NotConnectedRestException e) {
+            throw new NotConnectedException(e);
+        } catch (PermissionRestException e) {
+            throw new PermissionException(e);
+        }
+        List<TaskId> lTaskIds = new ArrayList<TaskId>(page.getList().size());
+        for (TaskStateData taskStateData : page.getList()) {
+            TaskInfoData taskInfo = taskStateData.getTaskInfo();
+            TaskIdData taskIdData = taskInfo.getTaskId();
+            JobId jobId = new JobIdImpl(taskInfo.getJobId().getId(), taskInfo.getJobId().getReadableName());
+            TaskId taskId = TaskIdImpl.createTaskId(jobId, taskIdData.getReadableName(), taskIdData.getId());
+            lTaskIds.add(taskId);
+        }
+        return new Page<TaskId>(lTaskIds, page.getSize());
+    }
+
+    @Override
+    public Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)